### PR TITLE
Jetty client two way TLS certificate authentication

### PR DESCRIPTION
**Pull request by stevenschlansker**

### CODE REVIEW

File: shared/src/main/java/com/opentable/jaxrs/TlsProvider.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.opentable.jaxrs;
+
+import java.security.KeyStore;
+import java.security.cert.CRL;
+
+public interface TlsProvider {
```
Comment (stevenschlansker) [inline shared/src/main/java/com/opentable/jaxrs/TlsProvider.java]: this interface serves to allow `otj-jaxrs` to link with `otj-credentials-client` without a compile time dependency.

File: clientfactory-resteasy/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java [hunk 2]
```diff
@@ -36,16 +39,28 @@
  */
 public class JaxRsClientFactoryImpl implements InternalClientFactory
 {
+    private Supplier<TlsProvider> provider; // NOPMD
 
     public JaxRsClientFactoryImpl(ApplicationContext ctx) {
 //        if (ctx != null && ClassUtils.isPresent("org.eclipse.jetty.server.Server", null)) {
             // TODO: figure out how to wire up thread pool
 //        }
```
Comment (pennello) [inline clientfactory-resteasy/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: Why is the formatting on these comments so messed up?

File: clientfactory-resteasy/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java [hunk 3]
```diff
@@ -36,16 +39,28 @@
  */
 public class JaxRsClientFactoryImpl implements InternalClientFactory
 {
+    private Supplier<TlsProvider> provider; // NOPMD
```
Comment (pennello) [inline clientfactory-resteasy/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: Which PMD is this complaining about?

Comment (stevenschlansker) [reply clientfactory-resteasy/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: good catch, this was vestigial.

File: clientfactory-resteasy/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java (hunk 2 above)
Comment (stevenschlansker) [reply clientfactory-resteasy/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: Not sure, but it's been like that for a while :/

**Pull request merged**